metadata_backend/api/middlewares.py

```diff
@@ -0,0 +1,50 @@
+"""Middleware methods for server."""
+import json
+from typing import Callable
+
+from aiohttp.web import HTTPError, Request, Response, middleware
+
+
+def error_middleware() -> Callable:
+    """Middleware for handling exceptions recieved from the API methods.
+
+    :returns: Handled response
+    """
+
+    @middleware
+    async def http_error_handler(req: Request, handler: Callable) -> Response:
+        """Handle HTTP errors.
+
+        :param req: A request instance
+        :param handler: A request handler
+        :returns: JSON response for the error
+        """
+        try:
+            response = await handler(req)
+            if response.status == 404:
+                return _json_exception(response.status,
+                                       Exception(response.message))
+            return response
+        except HTTPError as error:
+            return _json_exception(error.status, error)
+
+        # If something else goes wrong, convert to Internal Server Error
+        except Exception as ex:
+            return _json_exception(500, ex)
+
+    return http_error_handler
+
+
+def _json_exception(status: int, exception: Exception) -> Response:
+    """Convert an HTTP exception into JSON response.
+
+    :param status: Status code of the HTTP exception
+    :param exception: Exception content
+    :returns: Response in JSON format
+    """
+    body = json.dumps({
+        'error': exception.__class__.__name__,
```

Comment:
Maybe better to return error as key and reason as value? I don't think class name is important for user and correct status code is anyways always returned.

Reply:
we will change this in the OpenAPI specs. so fo now I think returning the HTTP status (as redundant it is) would suffice.